storage/innobase/include/sync0policy.ic

```diff
@@ -80,7 +80,7 @@ void MutexDebug<Mutex>::locked(
 	UNIV_NOTHROW
 {
 	ut_ad(!is_owned());
-	ut_ad(m_context.m_thread_id == os_thread_id_t(ULINT_UNDEFINED));
+	ut_ad(get_thread_id() == os_thread_id_t(ULINT_UNDEFINED));
```

Comment:
I believe these assertions should naturally go to ```::locked()```. This way we can avoid 3x mutex acquisition.